nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -379,9 +408,9 @@ def run(config):
         kwargs = dict(zip(kwargs_names, [SigOptImagenetTrainable,
                                          *tune.run.__defaults__]))
     else:
-        imagenet_trainable = config.get("imagenet_trainable", ImagenetTrainable)
-        assert issubclass(imagenet_trainable, ImagenetTrainable)
-        kwargs = dict(zip(kwargs_names, [imagenet_trainable, *tune.run.__defaults__]))
+        supervised_trainable = config.get("supervised_trainable", SupervisedTrainable)
```

Comment:
Why is the variable called supervised_trainable? Can we instead just call it trainable and then point to whatever trainable class the user wants?

Reply:
Good catch. Just updated.

Reply:
It is a matter of opinion, but I would still argue that it should be just `trainable` and not `ray_trainable`, since we discussed about removing explicit references to Ray whenever possible. 

We should be allowed to use other Trainables (considering the trainable is just a boilerplate function that call iterations from any class until a stopping criteria is met, and save the outcome at each step).

Reply:
I believe that strategy was discussed particularly for the Experiment classes. Here, the trainable will be passed to `ray.tune.run` so I don't think the same philosophy can be applied.